hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/BasicFormatterImpl.java

```diff
@@ -97,6 +97,19 @@ public String perform() {
 						lcToken = token;
 						misc();
 						break;
+					case "-":
+						if (lcToken.equals("-"))
+						{
+							String ttt;
+							do {
+								ttt = tokens.nextToken();
+								token += ttt;
+							}
+							while (tokens.hasMoreTokens());
+							lcToken = token;
+							misc();
+						}
+						break;
```

Comment:
It doesn't seem like your code distinguishes a simple minus sign `-` from a comment `--` properly. The following should work though:

```suggestion
						String ttt;
						if ( tokens.hasMoreTokens() && ( ttt = tokens.nextToken() ).equals( "-" ) ) {
							// Comment
							token += "--";
							while (tokens.hasMoreTokens()) {
								ttt = tokens.nextToken();
								token += ttt;
							}
							lcToken = token;
							misc();
						}
						else {
							misc();
						}
						break;
```

Reply:
If you look at the code you suggested, 
token += "--";
That part.
The -- code is ---, so that part... 
As a result of the test by changing to token += "-";, it is expressed normally and committed accordingly.